agent/api/task.go

```diff
@@ -314,6 +314,7 @@ func (task *Task) dockerHostConfig(container *Container, dockerContainerMap map[
 		Binds:        binds,
 		PortBindings: dockerPortMap,
 		VolumesFrom:  volumesFrom,
+		NetworkMode:  "host",
```

Comment:
The next statement may parse the `container.DockerConfig.HostConfig` string and override this value, do you know under which circumstances this would happen?